modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb

```diff
@@ -0,0 +1,945 @@
+
+
+          Exploit info:
+          - If you do not know how exploit for Windows 7/2008 work. Please read my exploit for Windows 7/2008 at
+          https://gist.github.com/worawit/bd04bad3cd231474763b873df081c09a because the trick for exploit is almost the same
+          - The exploit use heap of HAL for placing fake struct (address 0xffffffffffd00e00) and shellcode (address 0xffffffffffd01000).
+          On Windows 8 and Wndows 2012, the NX bit is set on this memory page. Need to disable it before controlling RIP.
+          - The exploit is likely to crash a target when it failed
+          - The overflow is happened on nonpaged pool so we need to massage target nonpaged pool.
+          - If exploit failed but target does not crash, try increasing 'GroomAllocations' value (at least 5)
+          - See the code and comment for exploit detail.
+
+
+          Disable NX method:
+          - The idea is from "Bypassing Windows 10 kernel ASLR (remote) by Stefan Le Berre" (see link in reference)
+          - The exploit is also the same but we need to trigger bug twice
+          - First trigger, set MDL.MappedSystemVa to target pte address
+          - Write '\x00' to disable the NX flag
+          - Second trigger, do the same as Windows 7 exploit
+          - From my test, if exploit disable NX successfully, I always get code execution
+        },
+        'Author' =>
+          [
+            'Equation Group',  # OG research and exploit
+            'Shadow Brokers',  # Hack and dump
+            'sleepya',         # Research and PoC
+            'wvu',             # Babby's first external module
+            'agalway-r7',      # External python module to internal ruby module (sorry wvu)
+            'cdelafuente-r7',  # ruby_smb wizard
+            'cdelafuente-r7'   # kernel debugging wizard
+          ],
+        'References' =>
+          [
+            ['msb', 'MS17-010'],
+            ['cve', '2017-0143'],
+            ['cve', '2017-0144'],
+            ['cve', '2017-0145'],
+            ['cve', '2017-0146'],
+            ['cve', '2017-0147'],
+            ['cve', '2017-0148'],
+            ['edb', '42030'],
+            ['url', 'https://github.com/worawit/MS17-010']
+          ],
+        'Date' => 'Mar 14 2017',
+        'Type' => 'remote_exploit',
+        'Rank' => 'average',
+        'Privileged' => true,
+        'wfsdelay' => 5,
+        'Targets' =>
+          [
+            [
+              'Windows 8.1, Windows 2012, Windows 10 Pro Build Server 2008 R2, and Windows 10 Enterprise Evaluation Build',
+              {
+                'Platform' => 'win',
+                'Arch' => [ARCH_X64],
+                'os_patterns' => ['Windows 2012', 'Windows 8.1', 'Windows 10 Pro Build', 'Windows 10 Enterprise Evaluation Build']
+              }
+            ]
+          ],
+        'notes' => {
+          AKA: ['ETERNALBLUE']
+        },
+        'DisclosureDate' => '2017-04-14'
```

Comment:
This should match the other modules. Either they will need to be updated or this should be made accurate.

```suggestion
        'DisclosureDate' => '2017-03-14'
```